make run-sonarqube fails on M1

Using the same context as in #360 

```
→ make run-sonarqube 
cd scripts && ./run-sonarqube.sh
Thu Dec 30 12:22:42 CET 2021
Run container using image tag 8.4-community
Error: No such container: ods-test-sonarqube
Thu Dec 30 12:22:42 CET 2021
[+] Building 1.4s (3/3) FINISHED                                                                                                                                                                                                          
 => [internal] load build definition from Dockerfile                                                                                                                                                                                 0.0s
 => => transferring dockerfile: 139B                                                                                                                                                                                                 0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                    0.0s
 => => transferring context: 2B                                                                                                                                                                                                      0.0s
 => ERROR [internal] load metadata for docker.io/library/sonarqube:8.4-community                                                                                                                                                     1.3s
------
 > [internal] load metadata for docker.io/library/sonarqube:8.4-community:
------
failed to solve with frontend dockerfile.v0: failed to create LLB definition: no match for platform in manifest sha256:a0aac11dcf18e237af1de207402e222d1e251ebc21b2bbcdac2b9b87710f1dd3: not found
make: *** [run-sonarqube] Error 1
                                                   
```

Reacting to the secomp warning tried out with a workaround inspired by https://stackoverflow.com/questions/47210302/sonarqube-6-7-failed-to-start-because-config-seccomp-not-compiled-into-kernel (see commit 1ba67723).

With this the container starts and no longer exits. But it is not providing the expected response either even if trying 30 minutes later via `scripts/waitfor-sonarqube.sh`.

Below is the log:
```
 → make run-sonarqube
cd scripts && ./run-sonarqube.sh
Thu Dec 30 12:47:28 CET 2021
Run container using image tag 8.4-community
ods-test-sonarqube
Thu Dec 30 12:47:28 CET 2021
[+] Building 1.4s (7/7) FINISHED                                                                                                                                                                                                          
 => [internal] load build definition from Dockerfile                                                                                                                                                                                 0.0s
 => => transferring dockerfile: 376B                                                                                                                                                                                                 0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                    0.0s
 => => transferring context: 2B                                                                                                                                                                                                      0.0s
 => [internal] load metadata for docker.io/library/sonarqube:8.4-community                                                                                                                                                           1.3s
 => [1/3] FROM docker.io/library/sonarqube:8.4-community@sha256:a0aac11dcf18e237af1de207402e222d1e251ebc21b2bbcdac2b9b87710f1dd3                                                                                                     0.0s
 => CACHED [2/3] RUN echo "sonar.forceAuthentication=true" >> conf/sonar.properties                                                                                                                                                  0.0s
 => CACHED [3/3] RUN echo "sonar.search.javaAdditionalOpts=-Dbootstrap.system_call_filter=false" >> conf/sonar.properties                                                                                                            0.0s
 => exporting to image                                                                                                                                                                                                               0.0s
 => => exporting layers                                                                                                                                                                                                              0.0s
 => => writing image sha256:7e2a3e8fcb4b419d7d041492c17fcd9f8c357b3a09638c99b5a7593936b0855a                                                                                                                                         0.0s
 => => naming to docker.io/library/ods-test-sonarqube                                                                                                                                                                                0.0s
Thu Dec 30 12:47:30 CET 2021
f0b7093d206789bfe858a032b2d4030e79dddc2bdeb1ba7c4bc4c15da1146cc5
Waiting up to 5 minutes for SonarQube to start ...
..............................SonarQube did not start, got health=.
2021.12.30 11:47:35 INFO  app[][o.s.a.AppFileSystem] Cleaning or creating temp directory /opt/sonarqube/temp
2021.12.30 11:47:35 INFO  app[][o.s.a.es.EsSettings] Elasticsearch listening on /127.0.0.1:9001
2021.12.30 11:47:36 INFO  app[][o.s.a.ProcessLauncherImpl] Launch process[[key='es', ipcIndex=1, logFilenamePrefix=es]] from [/opt/sonarqube/elasticsearch]: /opt/sonarqube/elasticsearch/bin/elasticsearch
2021.12.30 11:47:36 INFO  app[][o.s.a.SchedulerImpl] Waiting for Elasticsearch to be up and running
2021.12.30 11:47:37 INFO  app[][o.e.p.PluginsService] no modules loaded
2021.12.30 11:47:37 INFO  app[][o.e.p.PluginsService] loaded plugin [org.elasticsearch.transport.Netty4Plugin]
OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
2021.12.30 11:47:51 INFO  es[][o.e.e.NodeEnvironment] using [1] data paths, mounts [[/ (overlay)]], net usable_space [21.9gb], net total_space [58.4gb], types [overlay]
2021.12.30 11:47:51 INFO  es[][o.e.e.NodeEnvironment] heap size [494.9mb], compressed ordinary object pointers [true]
2021.12.30 11:47:51 INFO  es[][o.e.n.Node] node name [sonarqube], node ID [7Dl-vlbzSzifdiVY2s2ZpQ]
2021.12.30 11:47:51 INFO  es[][o.e.n.Node] version[6.8.4], pid[74], build[default/tar/bca0c8d/2019-10-16T06:19:49.319352Z], OS[Linux/5.10.76-linuxkit/amd64], JVM[AdoptOpenJDK/OpenJDK 64-Bit Server VM/11.0.6/11.0.6+10]
2021.12.30 11:47:51 INFO  es[][o.e.n.Node] JVM arguments [-XX:+UseConcMarkSweepGC, -XX:CMSInitiatingOccupancyFraction=75, -XX:+UseCMSInitiatingOccupancyOnly, -Des.networkaddress.cache.ttl=60, -Des.networkaddress.cache.negative.ttl=10, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Djava.io.tmpdir=/opt/sonarqube/temp, -XX:ErrorFile=../logs/es_hs_err_pid%p.log, -Xmx512m, -Xms512m, -XX:+HeapDumpOnOutOfMemoryError, -Dbootstrap.system_call_filter=false, -Des.path.home=/opt/sonarqube/elasticsearch, -Des.path.conf=/opt/sonarqube/temp/conf/es, -Des.distribution.flavor=default, -Des.distribution.type=tar]
2021.12.30 11:47:55 INFO  es[][o.e.p.PluginsService] loaded module [analysis-common]
2021.12.30 11:47:55 INFO  es[][o.e.p.PluginsService] loaded module [lang-painless]
2021.12.30 11:47:55 INFO  es[][o.e.p.PluginsService] loaded module [mapper-extras]
2021.12.30 11:47:55 INFO  es[][o.e.p.PluginsService] loaded module [parent-join]
2021.12.30 11:47:55 INFO  es[][o.e.p.PluginsService] loaded module [percolator]
2021.12.30 11:47:55 INFO  es[][o.e.p.PluginsService] loaded module [reindex]
2021.12.30 11:47:55 INFO  es[][o.e.p.PluginsService] loaded module [repository-url]
2021.12.30 11:47:55 INFO  es[][o.e.p.PluginsService] loaded module [transport-netty4]
2021.12.30 11:47:55 INFO  es[][o.e.p.PluginsService] no plugins loaded
Thu Dec 30 12:52:32 CET 2021
make: *** [run-sonarqube] Error 1
                                                                                   
```

--- a/scripts/bitbucket/.gitignore
+++ b/scripts/bitbucket/.gitignore
@@ -1,0 +1,1 @@
+docker-atlassian-bitbucket-server/

--- a/scripts/run-bitbucket.sh
+++ b/scripts/run-bitbucket.sh
@@ -11,7 +11,6 @@ BITBUCKET_SERVER_HOST_PORT="7990"
 BITBUCKET_SERVER_CONTAINER_NAME="ods-test-bitbucket-server"
 BITBUCKET_SERVER_IMAGE_NAME="atlassian/bitbucket"
 BITBUCKET_SERVER_IMAGE_TAG="7.6.5"
-BITBUCKET_POSTGRES_HOST_PORT="5432"
 BITBUCKET_POSTGRES_CONTAINER_NAME="ods-test-bitbucket-postgres"
 BITBUCKET_POSTGRES_IMAGE_TAG="12"
 HELM_VALUES_FILE="${ODS_PIPELINE_DIR}/deploy/ods-pipeline/values.generated.yaml"
@@ -29,11 +28,29 @@ docker rm -f ${BITBUCKET_POSTGRES_CONTAINER_NAME} || true
 docker run  --name ${BITBUCKET_POSTGRES_CONTAINER_NAME} \
   -v "${ODS_PIPELINE_DIR}"/test/testdata/bitbucket-sql:/docker-entrypoint-initdb.d \
   -e POSTGRES_PASSWORD=jellyfish -e POSTGRES_USER=bitbucketuser -e POSTGRES_DB=bitbucket \
-  -d --net kind -p "${BITBUCKET_POSTGRES_HOST_PORT}:5432" \
+  -d --net kind \
   postgres:${BITBUCKET_POSTGRES_IMAGE_TAG}
 
 echo "Run Bitbucket Server pointing to Postgres"
 docker rm -f ${BITBUCKET_SERVER_CONTAINER_NAME} || true
+
+cd "${SCRIPT_DIR}"/bitbucket
+
+if [ "$(uname -m)" = "arm64" ]; then
+  BITBUCKET_SERVER_ARM_IMAGE_NAME="ods-test-bitbucket-arm"
+  if [ "$(docker images -q ${BITBUCKET_SERVER_ARM_IMAGE_NAME}:${BITBUCKET_SERVER_IMAGE_TAG} 2> /dev/null)" == "" ]; then
+    echo "Building Bitbucket Server arm64 image ..."
+    rm -rf docker-atlassian-bitbucket-server || true
+    git clone --recurse-submodules https://bitbucket.org/atlassian-docker/docker-atlassian-bitbucket-server.git
+    cd docker-atlassian-bitbucket-server
+    git checkout 0e57b62 # Last known working Git commit (no branches / tags available)
+    docker build -t ${BITBUCKET_SERVER_ARM_IMAGE_NAME}:${BITBUCKET_SERVER_IMAGE_TAG} --build-arg BITBUCKET_VERSION=${BITBUCKET_SERVER_IMAGE_TAG} .
+  else
+    echo "Using existing ${BITBUCKET_SERVER_ARM_IMAGE_NAME}:${BITBUCKET_SERVER_IMAGE_TAG} image"
+  fi
+  BITBUCKET_SERVER_IMAGE_NAME="${BITBUCKET_SERVER_ARM_IMAGE_NAME}"
+fi
+
 docker run --name ${BITBUCKET_SERVER_CONTAINER_NAME} \
   -e JDBC_DRIVER=org.postgresql.Driver \
   -e JDBC_USER=bitbucketuser \

--- a/scripts/run-sonarqube.sh
+++ b/scripts/run-sonarqube.sh
@@ -27,10 +27,30 @@ esac; shift; done
 
 echo "Run container using image tag ${SONAR_IMAGE_TAG}"
 docker rm -f ${CONTAINER_NAME} || true
+
 cd "${SCRIPT_DIR}"/sonarqube
-docker build -t ${IMAGE_NAME} .
+
+if [ "$(uname -m)" = "arm64" ]; then
+    SONAR_ARM_IMAGE_NAME="ods-test-sonarqube-arm"
+    if [ "$(docker images -q ${SONAR_ARM_IMAGE_NAME}:${SONAR_IMAGE_TAG} 2> /dev/null)" == "" ]; then
+        echo "Building SonarQube arm64 image ..."
+        rm -rf docker-sonarqube || true
+        git clone https://github.com/SonarSource/docker-sonarqube
+        cd docker-sonarqube
+        git checkout refs/tags/9.4.0 # Last available Git tag
+        cd 9/community
+        docker build -t sonarqube-arm:${SONAR_IMAGE_TAG} .
+        cd "${SCRIPT_DIR}"/sonarqube
+        docker build -t ${SONAR_ARM_IMAGE_NAME}:${SONAR_IMAGE_TAG} --build-arg=from=sonarqube-arm:${SONAR_IMAGE_TAG} .
+    else
+        echo "Using existing ${SONAR_ARM_IMAGE_NAME}:${SONAR_IMAGE_TAG} image"
+    fi
+    IMAGE_NAME="${SONAR_ARM_IMAGE_NAME}"
+else
+    docker build -t ${IMAGE_NAME}:${SONAR_IMAGE_TAG} --build-arg=from=sonarqube:${SONAR_IMAGE_TAG} .
+fi
 cd - &> /dev/null
-docker run -d --net kind --name ${CONTAINER_NAME} -e SONAR_ES_BOOTSTRAP_CHECKS_DISABLE=true -p "${HOST_PORT}:9000" ${IMAGE_NAME}
+docker run -d --net kind --name ${CONTAINER_NAME} -e SONAR_ES_BOOTSTRAP_CHECKS_DISABLE=true -p "${HOST_PORT}:9000" ${IMAGE_NAME}:${SONAR_IMAGE_TAG}
 
 SONARQUBE_URL="http://localhost:${HOST_PORT}"
 if ! "${SCRIPT_DIR}/waitfor-sonarqube.sh" ; then

--- a/scripts/sonarqube/.gitignore
+++ b/scripts/sonarqube/.gitignore
@@ -1,0 +1,1 @@
+docker-sonarqube/

--- a/scripts/sonarqube/Dockerfile
+++ b/scripts/sonarqube/Dockerfile
@@ -1,3 +1,8 @@
-FROM sonarqube:8.4-community
+ARG from=""
+FROM $from
 
 RUN echo "sonar.forceAuthentication=true" >> conf/sonar.properties
+
+# Uncomment the following line to see debug output, e.g. in case SonarQube is
+# not starting up properly.
+# RUN echo "sonar.log.level=DEBUG" >> conf/sonar.properties

